pycbc/events/cuts.py

```diff
@@ -0,0 +1,336 @@
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+
+"""
+This module contains functions for reading in command line options and
+applying cuts to triggers or templates in the offline search
+"""
+import logging
+import copy
+import numpy as np
+from pycbc.events import ranking
+from pycbc.io import hdf
+from pycbc.bank import conversions as bank_conv
+from pycbc.io import get_chisq_from_file_choice
+
+sngl_rank_keys = ranking.sngls_ranking_function_dict.keys()
+
+trigger_param_choices = list(sngl_rank_keys)
+trigger_param_choices += [cc + '_chisq' for cc in hdf.chisq_choices]
+trigger_param_choices += ['end_time', 'psd_var_val', 'sigmasq']
+
+template_fit_param_choices = ['fit_by_fit_coeff', 'smoothed_fit_coeff',
+                              'fit_by_count_above_thresh',
+                              'smoothed_fit_count_above_thresh',
+                              'fit_by_count_in_template',
+                              'smoothed_fit_count_in_template']
+template_param_choices = bank_conv.conversion_options + \
+                             template_fit_param_choices
+
+ineq_choices = ['upper', 'lower', 'upper_inc', 'lower_inc']
+
+
+def insert_cuts_option_group(parser):
+    """
+    Add options to the parser for cuts to the templates/triggers
+    """
+    parser.add_argument('--trigger-cuts', nargs='+',
+                        help="Cuts to apply to the triggers, supplied as "
+                             "PARAMETER:VALUE:LIMIT, where, PARAMETER is the "
+                             "parameter to be cut, VALUE is the value at "
+                             "which it is cut, which must be convertible to "
+                             "a float, and LIMIT is one of '"
+                             + "', '".join(ineq_choices) +
+                             "' to indicate the inequality needed. "
+                             "PARAMETER is one of:'"
+                             + "', '".join(trigger_param_choices) +
+                             "'. For example snr:6:LOWER places the "
+                             "requirement that matched filter SNR must be > 6")
```

Comment:
It's not completely clear if triggers with SNR>6 are retained, or are cut.
I suppose it means they are kept, so I suggest '.. removes triggers if matched filter SNR < 6'